packages/babel-parser/src/parser/statement.js

```diff
@@ -246,18 +239,22 @@ export default class StatementParser extends ExpressionParser {
     }
 
     if (this.match(tt._export)) {
-      if (allowExport) {
-        return;
-      } else {
+      if (!allowExport) {
+        this.unexpected();
+      }
+
+      if (
+        this.hasPlugin("decorators") &&
```

Comment:
Not a blocker for this PR, but it would be great if we could integrate this with the plugin handling logic in https://github.com/babel/babel/blob/6b91d6434d2682ef6b0bf4ff5c9540b8ddbcd7f9/packages/babel-core/src/transformation/normalize-file.js#L129 so that the message gave useful feedback about the Babel plugin instead of the parser plugin.